File: library_common/src/main/java/com/xxjy/common/util/GsonTool.java
//package com.xxjy.common.util;
//
//import android.content.Context;
//import android.content.res.AssetManager;
//
//import com.google.gson.Gson;
//import com.xxjy.jyyh.entity.picker.CityJsonBean;
//import com.xxjy.jyyh.entity.picker.JsonBean;
//
//import org.json.JSONArray;
//
//import java.io.BufferedReader;
//import java.io.IOException;
//import java.io.InputStreamReader;
//import java.util.ArrayList;
//
//public class GsonTool {
//
//    public static <T> T parseJson(String jsonData, Class<T> type) {
//        Gson gson = new Gson();
//        T result = gson.fromJson(jsonData, type);
//        return result;
//    }
//
//    public static String toJson(Object o) {
//        Gson gson = new Gson();
//        String json = gson.toJson(o);
//        return json;
//    }
//
//    public static String getJson(Context context, String fileName) {
//        StringBuilder stringBuilder = new StringBuilder();
//        try {
//            AssetManager assetManager = context.getAssets();
//            BufferedReader bf = new BufferedReader(new InputStreamReader(
//                    assetManager.open(fileName)));
//            String line;
//            while ((line = bf.readLine()) != null) {
//                stringBuilder.append(line);
//            }
//        } catch (IOException e) {
//            e.printStackTrace();
//        }
//        return stringBuilder.toString();
//    }
//
//    public static ArrayList<JsonBean> parseData(String result) {//Gson 解析
//        ArrayList<JsonBean> detail = new ArrayList<>();
//        try {
//            JSONArray data = new JSONArray(result);
//            Gson gson = new Gson();
//            for (int i = 0; i < data.length(); i++) {
//                JsonBean entity = gson.fromJson(data.optJSONObject(i).toString(), JsonBean.class);
//                detail.add(entity);
//            }
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
//        return detail;
//    }
//
//    public static ArrayList<CityJsonBean> parseCityData(String result) {    //Gson 解析
//        ArrayList<CityJsonBean> detail = new ArrayList<>();
//        try {
//            JSONArray data = new JSONArray(result);
//            Gson gson = new Gson();
//            for (int i = 0; i < data.length(); i++) {
//                CityJsonBean entity = gson.fromJson(data.optJSONObject(i).toString(), CityJsonBean.class);
//                detail.add(entity);
//            }
//        } catch (Exception e) {
//            e.printStackTrace();
//        }
//        return detail;
//    }
//}

File: library_common/src/main/java/com/xxjy/common/util/GsonTool.kt
package com.xxjy.common.util

import android.content.Context
import android.content.res.AssetManager
import com.google.gson.Gson
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

object GsonTool {
    fun <T> parseJson(jsonData: String?, type: Class<T>?): T {
        val gson = Gson()
        return gson.fromJson(jsonData, type)
    }

    fun toJson(o: Any?): String {
        val gson = Gson()
        return gson.toJson(o)
    }

    fun getJson(context: Context, fileName: String): String {
        val stringBuilder = StringBuilder()
        try {
            val assetManager: AssetManager = context.assets
            val bf = BufferedReader(
                InputStreamReader(
                    assetManager.open(fileName)
                )
            )
            var line: String?
            while (bf.readLine().also { line = it } != null) {
                stringBuilder.append(line)
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return stringBuilder.toString()
    }

//    fun parseData(result: String?): ArrayList<JsonBean> { //Gson 解析
//        val detail: ArrayList<JsonBean> = ArrayList<JsonBean>()
//        try {
//            val data = JSONArray(result)
//            val gson = Gson()
//            for (i in 0 until data.length()) {
//                val entity: JsonBean =
//                    gson.fromJson(data.optJSONObject(i).toString(), JsonBean::class.java)
//                detail.add(entity)
//            }
//        } catch (e: Exception) {
//            e.printStackTrace()
//        }
//        return detail
//    }
//
//    fun parseCityData(result: String?): ArrayList<CityJsonBean> {    //Gson 解析
//        val detail: ArrayList<CityJsonBean> = ArrayList<CityJsonBean>()
//        try {
//            val data = JSONArray(result)
//            val gson = Gson()
//            for (i in 0 until data.length()) {
//                val entity: CityJsonBean =
//                    gson.fromJson(data.optJSONObject(i).toString(), CityJsonBean::class.java)
//                detail.add(entity)
//            }
//        } catch (e: Exception) {
//            e.printStackTrace()
//        }
//        return detail
//    }
}